File: PLP_Java/src/Aula03_ListaDeExercicio/Q06/Q06_Car.java
package Aula03_ListaDeExercicio.Q06;

public class Q06_Car {
    private Q06_Motor motor;

    public Q06_Car(Q06_Motor motor) {
        this.motor = motor;
    }

    public void showCarDetails() {
        System.out.println("Carro com motor: " + motor.getType());
    }

    public static void main(String[] args) {
        Q06_Motor motor = new Q06_Motor("V8");
        Q06_Car car = new Q06_Car(motor);
        car.showCarDetails();
    }
}

File: PLP_python/Aula03_ListaDeExercicios(15)/Q06_Car.py
class Motor:
    def __init__(self, motor_type):
        self.motor_type = motor_type

class Car:
    def __init__(self, motor):
        self.motor = motor

    def show_car_details(self):
        """
        Displays the details of the car, specifically its motor type.
        """
        print("Carro com motor: {}".format(self.motor.motor_type))

if __name__ == "__main__":
    motor = Motor("V8")
    car = Car(motor)
    car.show_car_details()